x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -61,7 +63,8 @@
     private volatile long expirationTimeMillis;
     private final AtomicBoolean isCancelling = new AtomicBoolean(false);
 
-    private AtomicReference<MutableSearchResponse> searchResponse;
+    private final AtomicReference<MutableSearchResponse> searchResponse = new AtomicReference<>();
+    private final AtomicReference<Map<String, List<String>>> responseHeaders = new AtomicReference<>();
```

Comment:
instead of having these two atomic references, would it make sense to have the headers become part of MutableSearchResponse? I always worry about the complexity added by two independent multi-threaded data structures.

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/54349/commits/f6e8ab01a0564e371793d79a7d3dd145732783d0